pkg/controller/pxc/users.go

```diff
@@ -326,3 +326,11 @@ func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool
 func sha256Hash(data []byte) string {
 	return fmt.Sprintf("%x", sha256.Sum256(data))
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileSyncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady {
+		return nil
+	}
+
+	return r.syncPXCUsersWithProxySQL(cr)
```

Comment:
should we run it in a separate go routine?
to avoid locked main reconcile loop because of problems inside proxysql

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/433/commits/310bd35d6871fa9856d5f816d58f1a4a200a4f85

Reply:
I do not like that approach
1. We rise gorutine on every loop (which is expensive)
2. If proxysql hangs we make its life even worse, because it can't sync users  and we ask it to do in every next loop. So we just "DDoS" it.
If we do not want to wait too long I would do it like function with timeout.

Reply:
Eah, current go-routine-approach looks quite dodgy to me as well

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/433/commits/6938776a9caa734d7770ae1ef446ae91953998ac